### PR TITLE
test(coverage): data_registrar added CRUD validations for org-admin and data-custodian

### PR DESCRIPTION
**Pull request by gsantiago-orbis**
## TL;DR

Enhanced Data Channel Registrar tests and fixed type issues across multiple services.

## What changed?

**New on `data_channel_registrar`**

- Added comprehensive test coverage, including tests for user roles (org admin, data custodian)
- Created test utilities with mock user data and helper functions
- Fixed type definitions in the Data Channel Registrar and AuthX Token API
- Removed unnecessary console.log statements
- Added error handling for non-existent data channels
- Updated package dependencies to use catalog version for `@vitest/istanbul` for consistency
- Added test coverage configuration on `data_channel_registrar`
- Removed unnecesary cleanup of Durable Object in teardown. Causes memory leaks, and it is not necesarry when `isolatedStorage: true` in `vitest.config.ts`
  - https://developers.cloudflare.com/workers/testing/vitest-integration/known-issues/#isolated-storage

**Achieved so far 70% coverage**

## How to test?

1. Run the test suite with coverage:
```bash
cd apps/data_channel_registrar

pnpm test

pnpm test:coverage
```

2. Verify that tests pass for different user roles:
   - Org admin can list and read data channels but cannot create, update, or delete them
   - Data custodian can list, create, read, update, and delete data channels

## Why make this change?

This change improves the reliability and maintainability of the Data Channel Registrar service by:

1. Ensuring proper permission checks for `org-admin` and `data-custodian`
2. Adding comprehensive test coverage to catch regressions
3. Fixing type issues that could lead to runtime errors
4. Standardizing dependency versions across packages
5. Improving error handling for edge cases

The enhanced test suite provides better validation of the service's behavior, particularly around permission checks and data channel operations

### CODE REVIEW

**Comment by gsantiago-orbis** [comment]
* **#450** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst/450?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst/450?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `staging`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: apps/data_channel_registrar/test/integration.spec.ts [hunk 1]
```diff
@@ -19,118 +19,275 @@ function generateDataChannels(count: number = 5): DataChannel[] {
...
+      expect(listDataChannels.success).toBe(true);
+      expect(listDataChannels.data).toBeDefined();
+      expect(listDataChannels.data.length).toBe(0);
+    });
+
+    it('should be able to list data channels', async () => {
+      // 1. setup step 1: create a data channel
+      const dataChannel = generateDataChannels()[0];
+      await custodianCreatesDataChannel(dataChannel); // only custodian can create a data channel
+      await custodianCreatesDataChannel(dataChannel); // only custodian can create a data channel
```
Comment (ihammers-orbis) [inline apps/data_channel_registrar/test/integration.spec.ts:84]: is there a reason this line exists twice?

**Review by ihammers-orbis — APPROVED** [review]
approved - though please add a comment for why that line I commented on exists twice, or fix if it shouldn't.

File: apps/data_channel_registrar/test/integration.spec.ts (hunk 1 above)
Comment (gsantiago-orbis) [reply apps/data_channel_registrar/test/integration.spec.ts:84]: no, fixed on top stack

**Comment by gsantiago-orbis** [comment]
### Merge activity

* **May 17, 3:11 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/orbisoperations/catalyst/450).
